src/ray/rpc/worker/core_worker_client_pool.cc

```diff
@@ -53,5 +53,21 @@ void CoreWorkerClientPool::Disconnect(ray::WorkerID id) {
   client_map_.erase(it);
 }
 
+void CoreWorkerClientPool::DisconnectAll() {
+  // disconnect
+  for (auto it : client_map_) {
+    auto conn = it->second;
+    rpc::ExitRequest request;
+    request.set_disconnect(true);
+    request.set_worker_id(it.first.Binary());
```

Comment:
here you are not setting the current worker id but the remote worker id and when remote worker receives the request, it tries to disconnect from the worker id in the request which is itself.